control_plane/attachment_service/src/reconciler.rs

```diff
@@ -400,15 +418,7 @@ impl Reconciler {
                     wanted_conf.generation = self.generation.into();
                     tracing::info!("Observed configuration requires update.");
                     self.location_config(node_id, wanted_conf, None).await?;
-                    if let Err(e) = self
-                        .compute_hook
-                        .notify(self.tenant_shard_id, node_id)
-                        .await
-                    {
-                        tracing::warn!(
-                            "Failed to notify compute of newly attached pageserver {node_id}: {e}"
-                        );
-                    }
+                    self.compute_notify().await?;
```

Comment:
It's a subjective one: correctness is satisfied either way:
- because we would eventually retry the overall reconciliation, anywhere we do this, it is safe to drop out
- because control plane availability issues should be transient, we may reasonably retry forever.

What I've ended up with is to retry forever in the context of live_migration, because we believe that the tenant is available and do not wish to make it unavailable by proceeding without notifying the compute.  In this context (the general case, where we aren't doing live migration), I propagate the error instead, so that we do not jam up all reconciliations if the control plane is unavailable.

Reply:
> What I've ended up with is to retry forever in the context of live_migration, because we believe that the tenant is available

Could it be that the assumption is incorrect and the tenant isn't actually available (thinking out loud)?